modules/phase_field/src/postprocessors/FeatureFloodCount.C

```diff
@@ -268,39 +274,133 @@ void FeatureFloodCount::communicateAndMerge()
 
   /**
-   * Each processor needs information from all other processors to create a complete
+   * Send the data from all processors to the root to create a complete
    * global feature map.
    */
-  _communicator.barrier();
-  Moose::perf_log.push("allgather_packed_range()", "FeatureFloodCount");
-  _communicator.allgather_packed_range((void *)(nullptr), send_buffers.begin(), send_buffers.end(),
-                                       std::back_inserter(recv_buffers));
-  Moose::perf_log.pop("allgather_packed_range()", "FeatureFloodCount");
+  _communicator.gather_packed_range(0, (void *)(nullptr), send_buffers.begin(), send_buffers.end(),
+                                    std::back_inserter(recv_buffers));
+
+  if (_is_master)
+  {
+    // The root process now needs to deserialize and merge all of the data
+    deserialize(recv_buffers);
+    recv_buffers.clear();
+
+    mergeSets(true);
+
+    /**
+     * Perform a sort to give a parallel unique sorting to the identified features.
+     * We use the "min_entity_id" inside each feature to assign it's position in the
+     * sorted vector.
+     */
+    std::sort(_feature_sets.begin(), _feature_sets.end());
+
+#ifndef NDEBUG
+    /**
+     * Sanity check. Now that we've sorted the flattened vector of features
+     * we need to make sure that the counts vector still lines up appropriately
+     * with each feature's _var_index.
+     */
+    unsigned int feature_offset = 0;
+    for (auto map_num = decltype(_maps_size)(0); map_num < _maps_size; ++map_num)
+    {
+      // Skip empty map checks
+      if (_feature_counts_per_map[map_num] == 0)
+        continue;
+
+      // Check the begin and end of the current range
+      auto range_front = feature_offset;
+      auto range_back = feature_offset + _feature_counts_per_map[map_num] - 1;
+
+      mooseAssert(range_front <= range_back && range_back < _feature_count, "Indexing error in feature sets");
+
+      if (!_single_map_mode && (_feature_sets[range_front]._var_idx != map_num || _feature_sets[range_back]._var_idx != map_num))
+        mooseError("Error in _feature_sets sorting, map index: " << map_num);
+
+      feature_offset += _feature_counts_per_map[map_num];
+    }
+#endif
+  }
+}
+
+void
+FeatureFloodCount::buildLocalToGlobalIndices(std::vector<unsigned int> & local_to_global_all, std::vector<int> & counts) const
+{
+  mooseAssert(_is_master, "This method must only be called on the root processor");
+
+  counts.resize(_n_procs, 0);
+  for (const auto & feature : _feature_sets)
+    for (const auto & local_index_pair : feature._orig_ids)
+          // local index                                              // rank
+      if (local_index_pair.second >= static_cast<unsigned int>(counts[local_index_pair.first]))
+        counts[local_index_pair.first] = local_index_pair.second + 1;
+
+  unsigned int globalsize = 0;
+  std::vector<int> offsets(_n_procs);
+  for (auto i = decltype(_n_procs)(0); i < _n_procs; ++i)
```

Comment:
Could change this to:

```C++
for (auto i = beginIndex(offsets); i < offsets.size(); ++i)
```

Not 100% necessary - just trying to be consistent...